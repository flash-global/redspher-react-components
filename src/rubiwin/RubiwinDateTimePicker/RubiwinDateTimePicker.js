import React from 'react'
import PropTypes from 'prop-types'
import { DateTimePicker, MuiPickersUtilsProvider } from '@material-ui/pickers'
import DateFnsUtils from '@date-io/date-fns'
import { createMuiTheme, ThemeProvider } from '@material-ui/core'

const defaultMaterialTheme = createMuiTheme({
  palette: {
    primary: { main: '#00c3ff' }
  }
})

/** full documentation : https://material-ui-pickers.dev/api/DateTimePicker */
const RubiwinDateTimePicker = ({
  onChange,
  value,
  allowKeyboardControl = true,
  ampm = false,
  disabled = false,
  disableFuture = false,
  disablePast = false,
  emptyLabel = '',
  format = 'dd/MM/yyyy HH:mm',
  label = '',
  maxDate = new Date('2100-01-01'),
  minDate = new Date('1900-01-01'),
  onClose = () => {},
  onError = () => {},
  className = '',
  variant = 'dialog',
  TextFieldComponent,
  ...props
}) => (
  <MuiPickersUtilsProvider utils={DateFnsUtils}>
    <ThemeProvider theme={defaultMaterialTheme}>
      <DateTimePicker
        label={label}
        value={value}
        onChange={onChange}
        allowKeyboardControl={allowKeyboardControl}
        ampm={ampm}
        disabled={disabled}
        disableFuture={disableFuture}
        disablePast={disablePast}
        emptyLabel={emptyLabel}
        format={format}
        className={className}
        minDate={minDate}
        maxDate={maxDate}
        onClose={onClose}
        onError={onError}
        variant={variant}
        TextFieldComponent={TextFieldComponent}
        {...props}
      />
    </ThemeProvider>
  </MuiPickersUtilsProvider>
)

RubiwinDateTimePicker.propTypes = {
  /** onChange callback */
  onChange: PropTypes.func.isRequired,
  /** parsable date */
  value: PropTypes.oneOfType([
    PropTypes.instanceOf(Date).isRequired,
    PropTypes.string
  ]),
  /** Enables keyboard listener for moving between days in calendar */
  allowKeyboardControl: PropTypes.bool,
  /** 12h/24h view for hour selection clock */
  ampm: PropTypes.bool,
  /** add a className */
  className: PropTypes.string,
  /** Disable picker and text field */
  disabled: PropTypes.bool,
  /** Disable future dates */
  disableFuture: PropTypes.bool,
  /** Disable past dates */
  disablePast: PropTypes.bool,
  /** Message displaying in text field, if null passed (doesn't work in keyboard mode) */
  emptyLabel: PropTypes.string,
  /** Format string */
  format: PropTypes.string,
  /** label text */
  label: PropTypes.string,
  /** Max selectable date */
  maxDate: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
  /** Min selectable date */
  minDate: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
  /** On close callback */
  onClose: PropTypes.func,
  /** Picker container option */
  variant: PropTypes.string,
  /** on error callback */
  onError: PropTypes.func
}

export default RubiwinDateTimePicker
