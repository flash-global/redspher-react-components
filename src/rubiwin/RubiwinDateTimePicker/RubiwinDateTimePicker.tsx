import * as React from 'react';
import { TextField } from '@mui/material';
import { DateTimePicker, LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import { DateTimePickerProps } from '@mui/x-date-pickers/DateTimePicker/DateTimePicker';

interface IrubiwinDateTimePicker
  extends Omit<DateTimePickerProps<any, any>, 'renderInput'>,
    React.RefAttributes<HTMLDivElement> {
  ampm?: boolean;
  className?: string;
  disabled?: boolean;
  disableFuture?: boolean;
  disablePast?: boolean;
  format?: string;
  label?: string;
  maxDate?: Date | string;
  minDate?: Date | string;
  locale?: string | object;
}

/**
 * This is a Mui DateTimePicker branded for Rubiwin
 *
 * Demos:
 * - [Date Time Picker](https://mui.com/x/react-date-pickers/date-time-picker/)
 * - [Pickers](https://mui.com/x/react-date-pickers/)
 *
 * API:
 * - [DateTimePicker API](https://mui.com/x/api/date-pickers/date-time-picker/)
 */
const RubiwinDateTimePicker = ({
  ampm = false,
  disabled = false,
  disableFuture = false,
  disablePast = false,
  format = 'dd/MM/yyyy HH:mm',
  label = '',
  maxDate = new Date('2100-01-01'),
  minDate = new Date('1900-01-01'),
  className = '',
  locale,
  ...props
}: IrubiwinDateTimePicker) => (
  <LocalizationProvider adapterLocale={locale} dateAdapter={AdapterDateFns}>
    <DateTimePicker
      label={label}
      ampm={ampm}
      disabled={disabled}
      disableFuture={disableFuture}
      disablePast={disablePast}
      inputFormat={format}
      className={className}
      minDate={minDate}
      maxDate={maxDate}
      renderInput={(props) => <TextField {...props} variant='standard' />}
      {...props}
    />
  </LocalizationProvider>
);

export default RubiwinDateTimePicker;
