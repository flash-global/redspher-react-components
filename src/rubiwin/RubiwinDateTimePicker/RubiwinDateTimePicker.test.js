import React from 'react'
import renderer from 'react-test-renderer'
import RubiwinDateTimePicker from './RubiwinDateTimePicker'

const testProps = {
  label: 'test',
  value: new Date('2018-01-01T00:00:00.000Z').toISOString(),
  onChange: () => {},
  emptyLabel: 'select a date',
  className: 'testClass'
}

it('renders correctly props object', () => {
  const tree = renderer
    .create(<RubiwinDateTimePicker {...testProps} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
