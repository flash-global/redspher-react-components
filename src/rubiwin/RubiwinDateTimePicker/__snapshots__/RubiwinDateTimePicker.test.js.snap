// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly props object 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-5qcp56-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="mui-1"
    id="mui-1-label"
  >
    test
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-188x4da-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      aria-label="Choose date, selected date is Jan 1, 2018"
      aria-readonly={true}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="mui-1"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      readOnly={true}
      required={false}
      type="text"
      value="01/01/2018 01:00"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-173wfxe"
      >
        <span>
          test
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
